### DISCUSSION TITLE
Yellow borders when capturing the game

### DISCUSSION THREAD

**Discussion by AngelCMHxD** [ppy/osu#27626] [Q&A] [answered]
### Type

Other

### Bug description

Once started capturing the game a yellow border appears around the screen. This happens when opening or focusing the game while streaming the game on Discord with it's new capture method or with OBS using a specific capture method.

May also happen with any apps that use this screen capture method exclusive to Windows.

OBS Capture Method:
Sources > New > Display Capture > Properties > Capture Method, Select Windows 10 (1903 and up).
![image](https://github.com/ppy/osu/assets/57822483/39999237-5c5c-40b0-b034-bffa5bc66579)

Discord's new capture method:
Settings > Voice & Video > Use our advanced technology to capture your screen.
![image](https://github.com/ppy/osu/assets/57822483/f253636b-fe5a-42ab-87ed-a27fb8366f42)

### Screenshots or videos

Preview of the bug:
![image](https://github.com/ppy/osu/assets/57822483/54093146-cbc5-4299-92d7-57b3b85389fb)

Persists after closing the game:
![image](https://github.com/ppy/osu/assets/57822483/01c56637-c924-4a05-afda-cd37560d4cfd)

### Version

2024.312.1-lazer

### Logs

Logs:
[compressed-logs.zip](https://github.com/ppy/osu/files/14622228/compressed-logs.zip)

**Comment by TheBrunoRM** [comment]
Can effectively reproduce.

**Comment by longnguyen2004** [comment]
This is a Windows thing, to indicate that something is recording the screen.

**Reply by AngelCMHxD** [reply]
Yes, however you can give the permission to apps that capture your screen to not show any borders using these settings: (AFAIK only available on Windows 11 and turned on by default)
![196761638-bbb95721-421c-4e51-9a0b-a3e7233f1ab3~2](https://github.com/ppy/osu/assets/57822483/2112bb06-057d-47d5-8165-babe52a104fd)

And it does work as these yellow borders never show up *until* you open lazer, it doesn't happen when capturing anything else, even other fullscreen apps and games, only happens when opening lazer while capturing the screen and these do not go away until you close the game, stop capturing the screen and start capturing the screen again, but they will appear again once you open lazer.

**Reply by peppy** [reply, marked as answer]
sounds like a windows bug. please report to microsoft.

**Reply by AngelCMHxD** [reply]
> sounds like a windows bug. please report to microsoft.

If it happened with other apps, sure. But even if it's only happening with lazer? it's a windows-exclusive bug, but it's rather lazer-specific. I have already tried reproducing it with other apps and games (none of the 150+ apps/games that I have streamed or recorded ever showed these yellow borders), and with other recording apps that use this capture method and it really only happens with lazer, as long as I don't open lazer while I'm recording there aren't any yellow borders anywhere.

I don't find it appropriate for windows devs to get a bug report for something that happens only with one game.

**Reply by peppy** [reply]
sure, but it's not something we can easily fix, nor something we'd consider investigating due to low priority / minimal effect on users.

i can't repro on windows 11 either.

**Reply by AngelCMHxD** [reply]
unable to repro it's weird as i tested it in 4 different windows installations, which i was able to reproduce it in all of them, who knows

closing this as it's not game-breaking and has some workarounds, thanks for your reply and looking into it though :D

**Comment by GhostTales** [comment]
only fix i can see for this is playing osu lazer in borderless and not fullscreen

**Reply by AngelCMHxD** [reply]
Capturing only the lazer window instead of the whole screen also prevents it from happening, however that also adds the inconvenience of switching the source when you finish playing osu or want to show something else. If you don't care about the yellow borders and want to capture the whole screen you should know that the yellow borders don't (or shouldn't) show up on the stream/final recording :P

**Comment by FeringaMind** [comment]
I was also able to reproduce the exact same error described on my machine running Win 11. I was able to essentially fix it by switching the OSU! renderer from Direct3D 11 to OpenGL. Hope this helps
